src/win/process.c

```diff
@@ -1250,7 +1252,68 @@ static int uv__kill(HANDLE process_handle, int signum) {
     return UV_EINVAL;
   }
 
+  // Create a dump file for the targeted process, if the registry key
+  // `HKLM:Software\Microsoft\Windows\Windows Error Reporting\LocalDumps\DumpFolder`
+  // points to a valid folder.  The default value if the registry key does not
+  // exist is `%LOCALAPPDATA%\CrashDumps`, see [0] for more detail.
+  // [0]: https://learn.microsoft.com/en-us/windows/win32/wer/collecting-user-mode-dumps
+  if (signum == SIGQUIT) {
+    // Get process name
+    WCHAR basename[MAX_PATH];
+    GetModuleBaseNameW(process_handle, NULL, &basename[0], sizeof(basename));
+
+    // Get PID
+    DWORD pid = GetProcessId(process_handle);
+
+    // Get LocalDumps directory path:
+    HKEY registry_key;
+    WCHAR dump_folder[MAX_PATH];
+    DWORD ret = RegOpenKeyExW(HKEY_LOCAL_MACHINE,
+                  L"SOFTWARE\\Microsoft\\Windows\\Windows Error Reporting\\LocalDumps",
+                  0,
+                  KEY_QUERY_VALUE,
+                  &registry_key);
+    if (ret == ERROR_SUCCESS) {
+      DWORD dump_folder_len = sizeof(dump_folder);
+      DWORD key_type = 0;
+      ret = RegGetValueW(registry_key,
+                          NULL,
+                          L"DumpFolder",
+                          RRF_RT_ANY,
+                          &key_type,
+                          (PVOID) dump_folder,
+                          &dump_folder_len);
+      if (ret != ERROR_SUCCESS) {
+        // Default value for `dump_folder` is `%LOCALAPPDATA%\CrashDumps`
+        WCHAR * localappdata;
+        SHGetKnownFolderPath(&FOLDERID_LocalAppData, 0, NULL, &localappdata);
+        swprintf_s(dump_folder, sizeof(dump_folder), L"%ls\\CrashDumps", localappdata);
```

Comment:
check return values of `swprintf_s` to make sure it fit?

Reply:
Also, call it `_snwprintf`

Reply:
I don't think checking the return value is useful, since if it didn't fit, we will just fail the `CreateFile` later on, right?

Reply:
oh, that might require calling `_snwprintf_s` then?

Reply:
there are far too many ways to call this function on windows, and I don't know which one is right, other than attempting to see what was used before

Reply:
once upon a time...

commit c6ecf97aafc858c2ad1089fb78da6c586d61d8b6
Author: Ben Noordhuis <info@bnoordhuis.nl>
Date:   Fri Nov 8 12:22:37 2013 +0100

    windows: use _snwprintf(), not swprintf()
    
    Drop the _CRT_NON_CONFORMING_SWPRINTFS hack and just use _snwprintf().
    
    It's a long and complicated story but the gist of it is that the MS CRT
    had a swprintf() function before ISO C did, with a different function
    prototype to boot: the ISO C one takes a |size| argument, the MS one
    does not.
    
    The function prototype that's exported by mingw and mingw-w64 depends
    on the mingw version and the _CRT_NON_CONFORMING_SWPRINTFS define.
    If they don't match up, you get the wrong prototype and things will
    crash at run-time.
    
    Reduce the phase space by sidestepping the whole issue: drop swprintf()
    altogether and use _snwprintf() from now on.
    
    Fixes #990.